File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> map;
        int n=nums.size();
        int count=0;
        int prefix=0;
        map[0]=1; //empty subarray
        
        for(int i=0; i<n; i++){
            prefix+=nums[i];
            if(map.find(prefix-k)!=map.end()){
                count+=map[prefix-k];
            }
            map[prefix]++;
        }
        return count;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        hashmap ={0:1}
        prefixsum = 0
        count = 0
        for num in nums:
            prefixsum = num + prefixsum
            diff = prefixsum - k
            
            if diff in hashmap:
                count += hashmap[diff]
                
            hashmap[prefixsum] = hashmap.get(prefixsum,0) + 1

        return count


        